src/hugr/validate.rs

```diff
@@ -720,19 +849,31 @@ pub enum InterGraphEdgeError {
         from_offset: Port,
         typ: ClassicType,
     },
+    #[error("Two linked ports both have the same direction")]
```

Comment:
This is part of portgraph guarantees, it should be safe to `unwrap` it instead of propagating.

Reply:
In this case I don't think there's anything to unwrap - I need to fill the case or else rust will complain about it being missing, so I guess I could panic instead?

Reply:
Ahh, nice